### PR TITLE
feat(PLM-388): Support check name as keyword

### PR DESCRIPTION
**Pull request by kilikkuo**
[Why]
Would be easier to write the Checker

[How]
1. Provide keyword arguments in Checker constructor
2. Add some functions to ensure it is constructed correctly
3. Add test & warning messages for deprecation

[PLM-388]

Reason
--------------

Changes
--------------

Test Scope
--------------

Checks
--------------
- [x] Unit tests are included or not applicable.

[PLM-388]: https://hardcoretech.atlassian.net/browse/PLM-388?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/test_spec.py [hunk 1]
```diff
@@ -1036,11 +1176,53 @@ class GreaterThanSpec:
...
+
+    def test_repeated_check_keyword(self):
+        with self.assertRaises(TypeError):
+            Checker([STR], optional=True, OPTIONAL=False)
+
+        with self.assertRaises(TypeError):
+            Checker([STR], op=CheckerOP.ANY, OP=CheckerOP.ALL)
+
+        with self.assertRaises(TypeError):
+            Checker([STR], allow_none=True, ALLOW_NONE=False)
```
Comment (cmj0121) [inline test/test_spec.py]: nit: I think it is hard to identify the diff between blacklist and repeated cases

**Review by cmj0121 — APPROVED** [review]
Perfect!

File: test/test_spec.py (hunk 1 above)
Comment (kilikkuo) [reply test/test_spec.py]: yeah, you're right, I think I can either make the warning more informative or throw different kinds of errors. Will address it. thanks

**Comment by kilikkuo** [comment]
Thanks!

**Pull request merged**